### DISCUSSION TITLE
MultiSoftwareProject One Repo or More?

### DISCUSSION THREAD

**Discussion by ssgnoe** [community/community#22278] [New to GitHub]
<p>Hey.</p>
<p>I’m student right now, and I’d like to make a Software for some Clients. The thing is, the Software should be on Laravel and some parts are identically for every client. But each client will get its own version of this software and some features will be added to everyone.</p>
<p>So what do you think, should I make a GitRepo for each client? If I’ve got a Bug, I have to fix on every Repo… Or should I work with Branches in one Repo, what not easily works with Git Flow and I will have many Clientbranches, so it could be hard to manage all of them…</p>
<p>Thank you, if you can help me :)…</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/ssgnoe">@ssgnoe</a>,</p>
<p>Welcome to Github Community Forum!</p>
<p>The second one is the best one, you need to make sure to use the master branch for the software with the basic features and create branches/versions for every client. If you find a bug in one of the version you can solve in the master branch and once resolved you can merge master in the other branches or even cherry pick the commit where you solve that bug (this work only if the bug is solved in only one commit). Is not easy to do but with a good branch name and version you can do everything.</p>
<p>Another way to do what you ask is using submodules: <a href="https://git-scm.com/book/en/v2/Git-Tools-Submodules" rel="nofollow noopener">Git-Tools-Submodules</a>, <a href="https://gist.github.com/gitaarik/8735255" rel="nofollow noopener">github-gist-submodules</a>, <a href="https://github.blog/2016-02-01-working-with-submodules/" rel="nofollow noopener">github-blog-working-with-submodules</a>. It is the first method you explained, but with some changes, but maybe is much more easy like this and probably is the better way.</p>
<p>Hope to solve your problem :blush: </p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by ssgnoe** [comment]
<p>Thank you, you helped me!</p>
